queue_job/models/queue_job.py

```diff
@@ -308,13 +308,16 @@ def _change_job_state(self, state, result=None):
             if state == DONE:
                 job_.set_done(result=result)
                 job_.store()
+                record.env["queue.job"].flush()
```

Comment:
Why is this flush necessary? This isn't the code path that was changed.

Reply:
Hello @amh-mw , could you have a look at this PR:
- https://github.com/OCA/queue/pull/677

The parent jobs need to be flushed before updating the child. Like v16 as is ([here](https://github.com/OCA/queue/blob/7e5efe7263ba4f79f704b9abc7d565de8285511f/queue_job/models/queue_job.py#L323))